config/resource_suppressions.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+    "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+    "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+    
+      <!-- Suppressions inside test and it folders -->
+      <suppress checks="NewlineAtEndOfFile" files="(src[\\/]it[\\/].*rule731selfexplanatory[\\/]InputJavadocMethodCheck\.java|src[\\/]test[\\/].*Newline.*AtEndOfFile\.java|src[\\/]test[\\/].*astprinter[\\/]InputJavadocComment\.javadoc|src[\\/]test[\\/].*astprinter[\\/]InputJavadocWithError\.javadoc|src[\\/]test[\\/].*checks[\\/]InputEmptyFile\.txt|src[\\/]test[\\/].*grammars[\\/]javadoc[\\/]htmlTags[\\/].*|src[\\/]test[\\/].*grammars[\\/]javadoc[\\/]javadocTags[\\/].*|src[\\/]test[\\/].*grammars[\\/]InputRegressionEmptyAst\.txt|src[\\/]test[\\/].*grammars[\\/]javadoc[\\/]InputLeadingAsterisks\.txt|src[\\/]test[\\/].*grammars[\\/]comments[\\/]InputFullOfSinglelineComments.java|src[\\/]test[\\/].*grammars[\\/]InputEmbeddedNullChar\.java)"/>
```

Comment:
>If you mean all on one line in one big expression, then no for the reasons I stated.

The reason for this is clear to me. 

No i meant that comment for all similar types of suppressions. Eg
```xml
<!-- Input files for NewlineAtEndOfFileCheckTest-->
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]test[\\/].*Newline.*AtEndOfFile\.java"/>

<!-- Many files in these folder require no newline at the end and are suppressed by default -->
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]test[\\/].*grammars[\\/]javadoc[\\/]htmlTags[\\/].*"/>
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]test[\\/].*grammars[\\/]javadoc[\\/]javadocTags[\\/].*"/>

<!--CI fails on windows system for this file -->
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]test[\\/].*grammars[\\/]InputEmbeddedNullChar\.java"/>

<!-- Some specific javadoc input files fail tests when they end with new line-->
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]test[\\/].*grammars[\\/]javadoc[\\/]InputLeadingAsterisks\.txt"/>
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]test[\\/].*grammars[\\/]comments[\\/]InputFullOfSinglelineComments\.java"/>
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]test[\\/].*astprinter[\\/]InputJavadocComment\.javadoc"/>
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]test[\\/].*astprinter[\\/]InputJavadocWithError\.javadoc"/>
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]it[\\/].*rule731selfexplanatory[\\/]InputJavadocMethodCheck\.java"/>

<!--Empty file cannot have any data  -->
      <suppress checks="NewlineAtEndOfFile" files="src[\\/]test[\\/].*checks[\\/]InputEmptyFile\.txt"/>
```